test/clientWithAutoConfirmEnabled.test.ts

```diff
@@ -51,6 +60,30 @@ test('signUp()', async () => {
   expect(user?.email).toBe(email)
 })
 
+test('setSession should return no error', async () => {
+  const { error, session } = await authWithSession.signUp({
+    email: setSessionEmail,
+    password,
+  })
+  expect(error).toBeNull()
+  expect(session).not.toBeNull()
+  await authWithSession.setSession(session?.refresh_token!)
+  const { user } = await authWithSession.update({ data: { hello: 'world' } })
+  expect(user?.user_metadata).toStrictEqual({ hello: 'world' })
+})
+
+test('setUser should update user in session', async () => {
+  const authSession = new GoTrueClient({
+    url: GOTRUE_URL,
+    autoRefreshToken: false,
+    persistSession: false,
+  })
+  const session = authWithSession.session()
+  const { user, error } = await authSession.setUser(session!.access_token)
+  expect(error).toBeNull()
+  expect(user?.user_metadata).toStrictEqual({ hello: 'world' })
```

Comment:
OK this is looking good. 

What happens if you call `authSession.update()` here?

Reply:
if we call `authSession.update()` here it will update the user data cause when you call `setUser()` it will update the session also. like this.
``` tsx
const session = { ...this.currentSession, access_token, token_type: 'bearer', user }
      this._saveSession(session)
``` 
so we have access_token in session and we need `access_token` to call `authSession.update()`